.circleci/config.yml

```diff
@@ -218,36 +237,121 @@ jobs:
           working_directory: ~/workbench/ui
           command: |
             yarn run codegen && yarn test --no-watch --no-progress --browsers=ChromeHeadless
+      - run:
+          name: Build with stricter compilation
+          working_directory: ~/workbench/ui
+          command: ./project.rb build --environment test
+
+  ui-deploy:
+    docker:
+      - image: allofustest/workbench:buildimage-0.0.11
+    working_directory: ~/workbench
+    steps:
+      - checkout
+      - run:
+          name: Fetch Submodules
+          command: git submodule update --init --recursive
+      - run:
+          working_directory: ~/workbench
+          name: Download Swagger CLI
+          command: |
+            ruby -r ./aou-utils/swagger.rb -e Workbench::Swagger.download_swagger_codegen_cli
+      - restore_cache:
+          keys:
+          - ui-cache-{{ checksum "~/workbench/ui/package.json" }}
+          - ui-cache-
+      - run:
+          working_directory: ~/workbench/ui
+          command: yarn install
+      - save_cache:
+          paths:
+            - ~/workbench/ui/node_modules
+          key: ui-cache-{{ checksum "~/workbench/ui/package.json" }}
       - deploy:
-          name: Maybe deploy to test/staging
+          name: Deploy to App Engine
           working_directory: ~/workbench/ui
           command: |
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
               ../ci/activate_creds.sh circle-sa-key.json
               ./project.rb deploy-ui \
                 --project all-of-us-workbench-test \
-                --account=circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+                --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
                 --version circle-ci-test \
                 --promote
             elif [ "${CIRCLE_TAG}" == "v*" ]; then
               ../ci/activate_creds.sh circle-sa-key.json
               ./project.rb deploy-ui \
                 --project all-of-us-rw-staging \
-                --account=circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+                --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
                 --version "${CIRCLE_TAG}" \
                 --promote
             else
-              echo "Not master branch or tagged release, skipping deploy; building only"
-              ./project.rb build --environment test
+              echo "Not master branch or tagged release, skipping deploy"
             fi
 
 workflows:
   version: 2
   build-test-deploy:
     jobs:
-      - api-build-test-deploy
-      - api-local-tests
-      - ui-build-test-deploy
+      # Always run basic test/lint/compilation (open PRs, master merge, tags).
+      - api-local-test:
+          filters:
+            tags:
+              only: /^v.*/
```

Comment:
I don't understand what this filter is doing. Can you add a comment?

Reply:
Done.